extensions/http-curl/processors/InvokeHTTP.cpp

```diff
@@ -202,15 +186,15 @@ bool InvokeHTTP::appendHeaders(const core::FlowFile& flow_file, /*std::invocable
   return true;
 }
 
-void InvokeHTTP::onTrigger(const std::shared_ptr<core::ProcessContext>& context, const std::shared_ptr<core::ProcessSession>& session) {
-  gsl_Expects(session && context && client_queue_);
+void InvokeHTTP::onTrigger(core::ProcessContext& context, core::ProcessSession& session) {
+  gsl_Expects(client_queue_);
 
-  auto flow_file = session->get();
+  auto flow_file = session.get();
 
   if (flow_file == nullptr) {
     if (!shouldEmitFlowFile()) {
-      logger_->log_debug("InvokeHTTP -- create flow file with {}", magic_enum::enum_name(method_));
-      flow_file = session->create();
+      logger_->log_debug("InvokeHTTP -- create flow file with  {}", std::string(magic_enum::enum_name(method_)));
```

Comment:
why do we need to construct and `std::string`? `enum_name` seems to be used without that in other logs